### PR TITLE
Fixing Up Subspace Tunneler

### PR DESCRIPTION
**Pull request by Shadowmech88**
Putting the nuke disk in a container no longer allows the subspace tunneler to hold it.
SMESs, PAs, AME parts, and the gateway are now on the prohibited list.
Fixed issue where contents are not ejected if the tunneler's `loc` is not a turf.
The destructive analyzer's attackby() now returns 1 so afterattack() isn't called on items inserted into it.
Mobs can now use the resist verb to escape from a tunneler if they're inside one.
Sparks will no longer be created if their loc would be null.

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -777,6 +777,38 @@ Thanks.
 
 	var/mob/living/L = usr
 
+	//Escaping from within a subspace tunneler.
```
Comment (clusterfack) [inline code/modules/mob/living/living.dm]: //WHY

Comment (Shadowmech88) [reply code/modules/mob/living/living.dm:780]: Is there a better way to identify whether the resisting mob is in the contents of a specific tunneler, or in the contents of another object that's in the contents of that tunneler?

Recursive contents checking was the only way I could think of.

Comment (clusterfack) [inline code/modules/mob/living/living.dm]: At this moment in time? Probably not, other codebases have designed better systems though.

Comment (Shadowmech88) [reply code/modules/mob/living/living.dm:780]: Is this bad enough that I should remove the ability to resist out of them?

Comment (Exxion) [reply code/modules/mob/living/living.dm:780]: Is there a particular reason you couldn't just use `get_holder_of_type()`?

Comment (Shadowmech88) [reply code/modules/mob/living/living.dm:780]: `get_holder_of_type()` seems to work just fine. I'll use that.

**Pull request merged**